### ISSUE TITLE
[core][autoscaler] Better observability for request resources 

### ISSUE THREAD

**Issue by rickyyx** [ray-project/ray#37959]
### What happened + What you expected to happen

Cluster constraint from requests resources should not be shown together with other delta demands (e.g. pending placement groups, pending tasks/acotrs) 

### Versions / Dependencies

master

### Reproduction script

NA

### Issue Severity

None

**Comment by ryanaoleary** [comment]
Feel free to assign this issue to me @kevin85421

**Comment by rickyyx** [comment]
Adding some more context in the original issue. 

I think this is more of a nit in letting users what's the current cluster resource constrain through `autoscaler_sdk.request_resources()`. For example in the `ray status [-v]` output, we should probably explicitly show what the cluster resource constraint is. 

However, in terms of priority, this is probably a good to have, like a P1.5.

**Comment by rueian** [comment]
+1 for this. In addition to the explicit `autoscaler_sdk.request_resources()`, I also want those reported backlogs to be explicitly visible from the dashboard. That will be very helpful when debugging issues like https://github.com/ray-project/ray/issues/48950 and https://github.com/ray-project/ray/issues/47531

**Comment by rueian** [comment]
WDYT @kevin85421 and @jjyao?

**Comment by rueian** [comment]
If that makes sense to you, I would like to take over this work.

**Comment by kevin85421** [comment]
@rueian I fully agree that the autoscaler on the dashboard would be very helpful. However, its scope is a bit broad and it is not the top priority for KubeRay v1.4.0—our top priority being stability—and the dashboard is currently undergoing significant refactoring by @MortalHappiness. I would recommend improving ray status for now and opening an issue to track progress on adding the autoscaler to the dashboard. Does it make sense to you?

**Comment by rueian** [comment]
No problem. Actually, the `ray status` and the dashboard share the same underlying information on this. So, we just need to augment that shared information, and there’s no need for any additional code changes on the dashboard to display the new information.

**Comment by rueian** [comment]
> Cluster constraint from requests resources should not be shown together with other delta demands (e.g. pending placement groups, pending tasks/acotrs)
> For example in the ray status [-v] output, we should probably explicitly show what the cluster resource constraint is.

To make cluster constraints clearer to users, I'd like to move them from the current `Demands` section to a new `Constraints` section in the output of `ray status`.

> I also want those reported backlogs to be explicitly visible from the dashboard. That will be very helpful when debugging issues like https://github.com/ray-project/ray/issues/48950 and https://github.com/ray-project/ray/issues/47531

I also like to introduce 3 counters, `num_ready_requests_queued`, `num_infeasible_requests_queued`, and `backlog_size` to the `Demands` section. I believe that will help us debug the leaking problems mentioned earlier.

So, I am going to make the result output look like this, for example:

```sh
Resources
---------------------------------------------------------------
Usage:
 0B/8.21GiB memory
 0B/2.00GiB object_store_memory

Constraints:
 {'CPU': 1}: 1000+ from request_resources()
Demands:
 {'CPU': 3.0}: 3+ pending tasks/actors
 {'CPU': 2.0}: 3+ pending tasks/actors
 2+ ready requests queued
 3+ infeasible requests queued
 1+ total backlog queued
```
@kevin85421, @jjyao does this look good to you?

**Comment by rueian** [comment]
Or this could be better:
```sh
Resources
---------------------------------------------------------------
Usage:
 0B/8.21GiB memory
 0B/2.00GiB object_store_memory

Constraints:
 {'CPU': 1}: 1000+ from request_resources()
Demands:
 {'CPU': 3.0}: 3+ pending tasks/actors
 {'CPU': 2.0}: 3+ pending tasks/actors
Demand Counts:
 2+ ready requests queued
 3+ infeasible requests queued
 1+ total backlog queued
```

**Comment by rueian** [comment]
ping @kevin85421, @jjyao. Does the change in the above new `ray status` look good to you?

**Comment by kevin85421** [comment]
@rueian Let's discuss offline.

**Comment by rueian** [comment]
As discussed offline with @kevin85421, I am going to do this in 2 PRs:
1. Addressing the original issue request to make the cluster constraints explicit in the report. This is ready for review https://github.com/ray-project/ray/pull/52278
2. Adding the `ready/infeasible/backlog` counts to the verbose report in the second PR.
